### DISCUSSION TITLE
All Contacts status or Stories

### DISCUSSION THREAD

**Discussion by oguzgu** [aldinokemal/go-whatsapp-web-multidevice#174] [Q&A]
Hi,

I couldn't find for getting statuses of my contacts. With status I mean stories part of whatsapp?
Is it possible to get statuses with this api?

**Comment by aldinokemal** [comment]
There is no dedicated stories/status endpoint, but status posts are not thrown away either. They arrive as normal messages on the chat JID `status@broadcast` and are stored like any other chat, so you can read them through the chat API added in `v7.10.0`:

```
GET /chats
GET /chat/status%40broadcast/messages?limit=50
```

Routes: `src/ui/rest/chat.go:21-22`. Storage has an explicit special case that titles that chat "Status" (`src/infrastructure/chatstorage/sqlite_repository.go:1622-1623`), and `CreateMessage` applies no broadcast filter (`src/infrastructure/chatstorage/sqlite_repository.go:1663`), so status posts land in the DB. Media on those messages can be pulled with `GET /message/:message_id/download` (`src/ui/rest/message.go:34`).

Two caveats:

1. Status posts are never forwarded to webhooks. That is a deliberate filter: `src/infrastructure/whatsapp/event_message_handler.go:167-172` drops any event whose source contains `broadcast`, so polling `/chat/status%40broadcast/messages` is the only way to see them.
2. The `status` field in `GET /user/info` is the text "about"/bio from WhatsApp (`src/usecase/user.go:100`), not stories. Different thing, easy to confuse.

Posting your own status is not implemented, there is no `/send/status` route (`src/ui/rest/send.go:16-27`). If you want that, a PR is welcome.

Note that whether your linked device actually receives a given contact's status depends on WhatsApp itself pushing it to companion devices, so it is worth checking the logs for messages with `status@broadcast` in the source before assuming it is an API problem.
